src/net_processing.cpp

```diff
@@ -5694,17 +5695,20 @@ bool PeerManagerImpl::SendMessages(CNode* pto)
                     LOCK(tx_relay->m_bloom_filter_mutex);
 
                     for (const auto& txinfo : vtxinfo) {
-                        const uint256& hash = peer->m_wtxid_relay ? txinfo.tx->GetWitnessHash() : txinfo.tx->GetHash();
-                        CInv inv(peer->m_wtxid_relay ? MSG_WTX : MSG_TX, hash);
-                        tx_relay->m_tx_inventory_to_send.erase(hash);
+                        CInv inv{MSG_TX, txinfo.tx->GetHash()};
+                        if (peer->m_wtxid_relay) {
+                            inv = CInv{MSG_WTX, txinfo.tx->GetWitnessHash()};
+                        }
```

Comment:
We need a different inv type `MSG_WTX` for wtxids!
Although the order could be switched (first create `MSG_WTX` and maybe overwrite that), since wtxid-based relay should be the standard by now and txid the exception.

Reply:
Switched the order